lib/parser.js

```diff
@@ -0,0 +1,507 @@
+/* eslint-disable one-var */
```

Comment:
this introduces an incredible amount of code that this library must now maintain. definitely not preferable. why is this necessary when moving to PostCSS@7?

Reply:
Because PostCSS 6+ uses different tokenizer. So `postcss/lib/parser` became completely different.

Reply:
OK. I would rather an update adapt to the changes in the tokenizer, or a compatible parser be made a separate module that others could use, than to introduce 507 new lines of code and a tokenizer to this module. There's already quite a barrier to outside contributors in this module, and the proposed addition is just going to make that barrier more difficult as well.

Reply:
OK, let's release it first and in next version you will rewrite parser when you have a time.

What do you think about it?

Reply:
I'm not sure "dangerous" is the right word there, seems awfully dramatic. I've suggested two alternatives:

- compatibility with the new tokenizer without the additional of a large amount if code
- a separate module that provides parser compat

If neither is acceptable, you may choose to fork and/or release an alternative. But I can't in good conscious accept a PR with changes that I don't think are wise.

Reply:
> a separate module that provides parser compat

How moving `lib/parser.js` to separated npm package will help?

Reply:
I'm not generally concerned about the size of `node_modules` - and I'm not part of the crowd that obsesses over that metric. Bundle sizes would not be effected unless poor choices were made in the separate package and/or tree-shaking was not used. That's not a compelling argument for including a 500 line parser in an otherwise small package.

Reply:
> That's not a compelling argument for including a 500 line parser in an otherwise small package.

So what use case will work worse with this 500 LOC? The last commit was at May 30, how many changes you plan to do so worry about code size?

We can’t rewrite the whole parser since it will require few days and you will not do it.

We can’t move forke the parser since it will force to change `dependencies` in too many projects.

We can’t keep it, since many plugins will not compatible with this parser (your parser generate nodes without necessary methods).

We can release `lib/parser.js` to separated npm package. But what it will change? What will be a difference if this code will be in `postcss-less-subparser`?

What solution do you wee as a maintainer?

Reply:
I'm not sure what you're trying to say. I see some things in there that might be trying to shame me into accepting the PR - not sure if that was the intent, but it's not really motivating.

> As a creator of PostCSS I need to take care of the ecosystem.

Sure.

> You are the only PostCSS 5 project right now.

And that's OK.

> We need to solve it somehow.

The parser that was being relied upon was drastically changed between PostCSS@5 and PostCSS@7. There needs to be a compatibility layer to bridge the gap. That layer doesn't belong in this module.

> I even spent my Sunday to fix it for you.

That's very generous of you. However, if you had started work and realized it would be a large change, it would have been prudent at that point to discuss a large change before spending the time on it - I'm pretty sure that's standard practice for most projects.

> I can spend few extra hours and remove code, Wich you override in Less parser to reduce 500 LOC to something around 300 LOC. Will it be OK?

I apologize - I don't know what you're trying to convey in that reply.

Reply:
> it's not really motivating.

Yeap, maybe I am overreacted.

But it is the same from your side. You just say “no” without really trying to find a way to solve it.
It is really not motivating me as well. It was really bad feedback for PR.

Let’s say sorry to each other and start again.

Here is my basement:

1. No one has time to rewrite the parser.
2. We need a release ASAP.

Here are solutions that I can see:

1. I can reduce the compatibility layer size. Like in `lib/less-parser` you override many methods, we can remove it.
2. We can release compatibility layer to npm package. But this package will not be supported very well since it will be used only in the single package.
3. You can give maintenance for me and I just release it as it.

What is your basement and what option do you see?